src/encoded/commands/deploy.py

```diff
@@ -194,15 +194,13 @@ def tag_ec2_instance(instance, tag_data, elasticsearch, cluster_name):
     return instance
 
 
-def read_ssh_key():
-    home = expanduser("~")
-    ssh_key_path = home + '/' + '.ssh/id_rsa.pub'
-    ssh_keygen_args = ['ssh-keygen', '-l', '-f', ssh_key_path]
+def read_ssh_key(identity_file):
+    ssh_keygen_args = ['ssh-keygen', '-l', '-f', identity_file]
```

Comment:
Where do you do some kind of validation on identify_file? You can only trust data an object owns and does not share.

Reply:
I got the answer. Validation is done implicitly in subprocess.check_output